treewide: fix bash exit handlers

erikarvstedt opened the pull request:
#### Copy of commit msg

Transform exit handlers of the form
```bash
trap cleanup EXIT [INT] [TERM] [QUIT] [HUP] [ERR]
  (where cleanup is idempotent)
```
to
```bash
trap cleanup EXIT
```

This fixes a common bash antipattern.

Each of the above signals causes the script to exit. For each signal, bash first handles the signal by running `cleanup` and then runs `cleanup` again when handling EXIT. 
(Exception:  `vscode/*` prevents the second run of `cleanup` by removing the trap in cleanup`).

Simplify the cleanup logic by just trapping exit, which is always run when the script exits due to any of the above signals.

Note: In case of borgbackup, the exit handler is not idempotent, but just trapping EXIT guarantees that it's only run once.

erikarvstedt committed:
treewide: fix bash exit handlers

Transform exit handlers of the form
trap cleanup EXIT [INT] [TERM] [QUIT] [HUP] [ERR]
  (where cleanup is idempotent)
to
trap cleanup EXIT

This fixes a common bash antipattern.

Each of the above signals causes the script to exit. For each signal,
bash first handles the signal by running `cleanup` and then runs
`cleanup` again when handling EXIT.
(Exception:  `vscode/*` prevents the second run of `cleanup` by removing
the trap in cleanup`).

Simplify the cleanup logic by just trapping exit, which is always run
when the script exits due to any of the above signals.

Note: In case of borgbackup, the exit handler is not idempotent, but just
trapping EXIT guarantees that it's only run once.

viraptor commented on pkgs/applications/editors/vscode/extensions/_maintainers/update-bin-srcs-lib.sh:
```diff
@@ -37,14 +37,12 @@ prefetchExtensionUnpacked() {
-    unset tmpDir
-    trap - INT TERM HUP EXIT
+    1>&2 printf "rm -rf %q\n" "$tmpDir"
+    rm -rf "$tmpDir"
```
Do you not want to preserve the end-of-options `--` here?

erikarvstedt replied:
No, we also don't do this at other places in nixpkgs. `tmpDir` is guaranteed to never begin with a `-`.

erikarvstedt replied:
I've extracted this change to a separate commit.

erikarvstedt committed:
vscode: simplify helper scripts

This simplifies the following commit.

- Remove `--` from `rm` because `$tmpDir` is guaranteed to never start
  with a `-`.
  We also don't use this pattern when removing tmpDirs at other places
  in nixpkgs.
- Remove `unset tmpDir` because the variable `tmpDir` is never
  accessed after the exit handler is run.
